compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -6395,6 +6396,195 @@ private void addStrandAnnotationWithThreadAny(BLangInvocation.BLangActionInvocat
                 .addAnnotation(this.strandAnnotAttachement.annotationAttachmentSymbol);
     }
 
+    @Override
+    public void visit(BLangInvocation.BLangResourceAccessInvocation resourceAccessInvocation) {
+        // `BLangResourceAccessInvocation` will be desugared into a `BLangInvocation`
+        // ex: 1
+        // Target resource function
+        // resource function get [string a]/[int b](string c) returns int {
+        //      return 3;
+        // }
+        //
+        // before desugar -
+        // a->/path/[1].get("hello")
+        //
+        // after desugar -
+        // a.get("path", 1, "hello")
+        //
+        // ex: 2
+        // Target resource function
+        // resource function get4 path/[int a]/[int... b](string c) returns int {
+        //      return 3;
+        // }
+        //
+        // before desugar -
+        // a->/path/[...list].get("hello")
+        //
+        // after desugar -
+        // a.get4(list[0], list.slice(1), "hello")
+        
+        if (resourceAccessInvocation.invokedInsideTransaction) {
+            transactionDesugar.startTransactionCoordinatorOnce(env, resourceAccessInvocation.pos);
+        }
+
+        // Create virtual invocation to reorder resource path parameters
+        BLangInvocation pathParamInvocation = createInvocationForPathParams(resourceAccessInvocation);
+        reorderArguments(pathParamInvocation);
+
+        BResourceFunction targetResourceFunc = resourceAccessInvocation.targetResourceFunc;
+        List<Name> resourcePath = targetResourceFunc.resourcePath;
+
+        int pathParamInvocationRequiredArgCount = pathParamInvocation.requiredArgs.size();
```

Comment:
Can't we just use `requiredArgCount`, instead of `pathParamInvocationRequiredArgCount`?

Reply:
R u asking whether we can rename `pathParamInvocationRequiredArgCount` to `requiredArgCount`?. I used the name `pathParamInvocationRequiredArgCount` since we have several invocations inside this block